### PR TITLE
Make `core.lifetime.forward` iterative instead of recursive

### PR DESCRIPTION
**Pull request by radcapricorn**
...using the new alias assignment feature

### CODE REVIEW

File: src/core/lifetime.d [hunk 1]
```diff
@@ -1556,15 +1555,16 @@ template forward(args...)
...
-        static if (args.length == 1)
-            alias forward = fwd;
-        else
-            alias forward = AliasSeq!(fwd, forward!(args[1..$]));
+            @property auto fwd(){ pragma(inline, true); return move(arg); }
     }
+
+    alias Result = AliasSeq!();
+    static foreach (arg; args)
+        Result = AliasSeq!(Result, fwd!arg);
```
Comment (maxhaton) [inline src/core/lifetime.d:1563]: Alias assign is new, does using it here lead to a bootstrapping issue?

Comment (ibuclaw) [reply src/core/lifetime.d:1563]: The host compiler doesn't build this.

**Comment by RazvanN7** [comment]
This seems to be passing all the tests, so I assume this is good to go?

**Comment by radcapricorn** [comment]
Eh?

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**